src/components/card-link/types.ts

```diff
@@ -0,0 +1,15 @@
+import { ReactNode } from 'react'
+import { CardProps } from 'components/card'
+
+/**
+ * NOTE: "base" is excluded from `elevation` because that elevation should only
+ * be used with non-interactive Cards, according to the design system.
+ */
+interface CardLinkProps {
+  children: ReactNode
+  className?: string
+  elevation?: Exclude<CardProps['elevation'], 'base'>
+  href: string
+}
```

Comment:
nit: Sometimes I find it helpful to add docs comments above properties if it makes sense. So for example, I might move the comment above the interface to right above `elevation`.

```suggestion
interface CardLinkProps {
  children: ReactNode
  className?: string
  /**
   * NOTE: "base" is excluded from `elevation` because that elevation should only
   * be used with non-interactive Cards, according to the design system.
   */
  elevation?: Exclude<CardProps['elevation'], 'base'>
  href: string
}
```

Reply:
So I thought about doing that, but went against it because while it still looks okay when there's just one comment above a property, I find it harder to read overall when every property has comments above it. Maybe we should try to establish some guidelines here? 🤔

Reply:
I'd say the comment above the property is preferable so that it shows up in intellisense:
![Screen Shot 2022-01-27 at 1 38 33 PM](https://user-images.githubusercontent.com/1289701/151431441-f55e72a7-8fd0-42b5-969b-ea82261ecb69.png)

Reply:
Great point, I've updated and added a comment for each property in https://github.com/hashicorp/dev-portal/pull/83/commits/b30dfa09ecf2a5af16035bfdabdde43b27cb68d0!